DentOS_Framework/DentOsTestbedLib/src/dent_os_testbed/lib/traffic/ixnetwork/ixnetwork_ixia_client_impl.py

```diff
@@ -663,3 +666,42 @@ def run_send_arp(self, device, command, *argv, **kwarg):
         return err, [{"success": msg["arg2"],
                       "port": msg["port"],
                       "src_ip": msg["src_ip"]} for msg in res]
+
+    def run_update_l1_config(self, device, command, *argv, **kwarg):
+        if not IxnetworkIxiaClientImpl.ixnet:
+            return 0, "Ixia not connected"
+        if command == "update_l1_config":
+            vports = IxnetworkIxiaClientImpl.ixnet.Vport.find()
+            ports = kwarg['params'][0].get("ixia_ports", [])
+            if len(ports) < 1:
+                raise ValueError(f" IXIA ports not provided")
+            # fd -> full duplex ; hd -> half duplex
+            duplex = "fd" if kwarg['params'][0].get("duplex", "Full").capitalize() == "Full" else "hd"
+            speed = kwarg['params'][0].get("speed", None)
+            if speed:
+                ixia_speed = self.__convert_to_ixia_speed(speed, duplex)
+            else:
+                ixia_speed = None
+            autoneg = kwarg['params'][0].get("autoneg", True)
+            names = [vport.Name for vport in vports]
+            for port in ports:
+                if port in names:
+                    required_ixia_port = vports[names.index(port)]
+                    card = required_ixia_port.L1Config.NovusTenGigLan or required_ixia_port.L1Config.Ethernet
+                    device.applog.info(f"Changing speed from {required_ixia_port.ActualSpeed} to {speed} \
+                                         on tgen_port {required_ixia_port.Name}")
+                    device.applog.info(f"Changing autoneg to {autoneg} on tgen_port {required_ixia_port.Name}")
+                    card.update(AutoNegotiate=autoneg, Speed=ixia_speed)
+            return 0, ""
+
+    @classmethod
+    def __convert_to_ixia_speed(self, speed, duplex):
+        if speed <= 100:
```

Comment:
"<=" -> `speed == 10 or speed == 100`
So you'll have complete validation of the speed value. Otherwise "99" is also less then 100 :)

Reply:
Mistake, thank you